File: src/main/java/me/tbg/match/bot/MatchBot.java
package me.tbg.match.bot;

import org.bukkit.plugin.java.JavaPlugin;

public class MatchBot extends JavaPlugin {

    private DiscordBot bot;
    private BotConfig config;

    @Override
    public void onEnable() {
        this.saveDefaultConfig();
        this.reloadConfig();
        this.config = new BotConfig(getConfig());
        this.bot = new DiscordBot(config, getLogger());
        this.registerListeners();
    }

    private void registerListeners() {
        this.getServer().getPluginManager().registerEvents(new MatchStartListener(bot), this);
        this.getServer().getPluginManager().registerEvents(new MatchFinishListener(bot), this);
    }

    public void reloadBotConfig() {
        this.reloadConfig();
        config.reload(getConfig());
        bot.reload();
    }
}

File: src/main/java/me/tbg/match/bot/MatchBot.kt
package me.tbg.match.bot

import org.bukkit.plugin.java.JavaPlugin

class MatchBot : JavaPlugin() {
    private var bot: DiscordBot? = null
    private var config: BotConfig? = null
    override fun onEnable() {
        saveDefaultConfig()
        reloadConfig()
        config = BotConfig(getConfig())
        bot = DiscordBot(config!!, logger)
        registerListeners()
    }

    private fun registerListeners() {
        server.pluginManager.registerEvents(MatchStartListener(bot), this)
        server.pluginManager.registerEvents(MatchFinishListener(bot!!), this)
    }

    fun reloadBotConfig() {
        reloadConfig()
        config!!.reload(getConfig())
        bot!!.reload()
    }
}